mautic/mautic
# Added support for postal email server

Postal is a complete and fully featured mail server for use by websites & web servers. Think Sendgrid, Mailgun or Postmark but open source and ready for you to run on your own servers.

| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | "features"
| Bug fix?                               | no
| New feature?                           | yes
| Deprecations?                          | no
| BC breaks?                             | no
| Automated tests included?              | yes/no
| Related user documentation PR URL      | mautic/mautic-documentation#... 

#### Description:

This PR adds a new mail transport for postal mail server. The transport extends the standard smtp transport but adds postal webhook handling  for failed deliveries and bounces.

Postal webhooks are documented here:
https://github.com/postalhq/postal/wiki/Webhook-Events-&-Payloads

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Setup a postal mail server
3. Configure a webhook in postal for your installation
4. Setup & send a campaign with an invalid email
5. Watch for failed mail delivery and check if contact is marked as bounced



<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/9754"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
